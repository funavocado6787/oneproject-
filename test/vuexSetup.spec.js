import { RouterLinkStub, shallowMount, createLocalVue, mount } from '@vue/test-utils'
import VuexSetup from '../pages/vuexSetup'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Vuetify)
import DisplayCode from '../components/displayCode.vue'
Vue.component(DisplayCode.name, DisplayCode);

const localVue = createLocalVue()
localVue.use(Vuex)

describe('VuexSetup.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            vuexSetupSnippets: {}
        }
        store = new Vuex.Store({
            modules: {
                codeSnippet: {
                    namespaced: true,
                    actions: {
                        loadVuexSetupSnippets({commit}) {
                            commit('setVuexSetupSnippets', {
                                vuex1: 'vuex1',
                                vuex2: 'vuex2',
                                vuex3: 'vuex3'
                             });
                        }
                    },
                    getters: {
                        getVuexSetupSnippets: state => state.vuexSetupSnippets
                    },
                    state,
                    mutations: {
                        setVuexSetupSnippets(state, data) {
                            state.vuexSetupSnippets = data;
                        }
                    }
                }
            }
        })
    })

    test('Smoke Test - check page renders by looking for some hard coded text', async () => {
        const wrapper = shallowMount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('State Management (Vuex)');
    })

    test('vuexSetup snippet load call was made on creation', async () => {
        const wrapper = shallowMount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('State Management (Vuex)');
        expect(Object.keys(state.vuexSetupSnippets).length).toBe(3);
    })

    test('vuexSetup made use of vuex1', async () => {
        const wrapper = mount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('vuex1');
    })

    test('vuexSetup made use of vuex2', async () => {
        const wrapper = mount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('vuex2');
    })

    test('vuexSetup made use of vuex3', async () => {
        const wrapper = mount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('vuex3');
    })

    test('match snapshot', async () => {
        const wrapper = mount(VuexSetup, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})













