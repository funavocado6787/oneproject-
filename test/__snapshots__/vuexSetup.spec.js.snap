// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VuexSetup.vue match snapshot 1`] = `
<div
  class="container layout column"
>
  <h3>
    State Management (Vuex)
  </h3>
   
  <div
    class="flex"
  >
    Install vuex via npm.
  </div>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor one-liner"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          vuex1
          <br />
        </pre>
      </div>
    </div>
  </div>
   
  <div
    class="flex pt-2"
  >
    Create a Vuex component.
  </div>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          vuex2
          <br />
        </pre>
      </div>
    </div>
  </div>
   
  <div
    class="flex pt-2"
  >
    
      Register the store with the root main app.js
  
  </div>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          vuex3
          <br />
        </pre>
      </div>
    </div>
  </div>
</div>
`;
